File: app/src/main/java/org/literacyapp/soundcards/util/TtsHelper.java
package org.literacyapp.soundcards.util;

import android.content.Context;
import android.speech.tts.TextToSpeech;
import android.util.Log;

import org.literacyapp.soundcards.SoundCardApplication;

public class TtsHelper {

    public static void speak(Context context, String text) {
        Log.i(TtsHelper.class.getName(), "speak");

        Log.i(TtsHelper.class.getName(), "text: " + text);

        SoundCardApplication application = (SoundCardApplication) context;
        TextToSpeech tts = application.getTts();
        tts.speak(text, TextToSpeech.QUEUE_FLUSH, null, null);
    }
}

File: app/src/main/java/ai/elimu/soundcards/util/TtsHelper.kt
package ai.elimu.soundcards.util

import ai.elimu.soundcards.SoundCardApplication
import android.content.Context
import android.speech.tts.TextToSpeech
import android.util.Log

object TtsHelper {
    fun speak(context: Context?, text: String?) {
        Log.i(TtsHelper::class.java.getName(), "speak")

        Log.i(TtsHelper::class.java.getName(), "text: " + text)

        val application = context as SoundCardApplication
        val tts = application.tts
        tts!!.speak(text, TextToSpeech.QUEUE_FLUSH, null, null)
    }
}
